crates/burn-jit/src/kernel/conv/conv2d/implicit_gemm.rs

```diff
@@ -0,0 +1,343 @@
+            let abs_slice_row = a_row + rel_slice_row; // Row of the matrix the slice is on
+
+            // Actual index within a slice (0 to `kernel_size * channels - 1`) that the thread is repsonsible for
+            let my_slice_idx = (slice_start_idx + (m % CMMA_K)) % slice_size;
+
+            // Given the row of the matrix that the slice is in, and the index of the thread
+            // within a slice, want to compute what input element to load...
+            // first compute coordinates in output space (center of the kernel in MxK matrix A)
+            let batch = abs_slice_row / batch_stride;
+            let out_y = (abs_slice_row % batch_stride) / y_stride;
+            let out_x = ((abs_slice_row % batch_stride) % y_stride) / x_stride;
+
+            let channel = my_slice_idx / kernel_size;
+
+            let kernel_y = (my_slice_idx / kernel_w) % kernel_h;
+            let kernel_x = my_slice_idx % kernel_w;
+            let y = (out_y * args.stride_h + kernel_y * args.dilation_h) as i32 - args.pad_h;
+            let x = (out_x * args.stride_w + kernel_x * args.dilation_w) as i32 - args.pad_w;
+
+            if x >= 0 && x < w && y >= 0 && y < h {
+                input_tile[m] = f16::cast_from(
+                    input[batch * input.stride(0)
+                        + y as u32 * input.stride(2)
+                        + x as u32 * input.stride(3)
+                        + channel * input.stride(1)],
+                );
+            } else {
+                input_tile[m] = f16::new(0.0);
+            }
+        }
+
+        /**************************** Loading Weight Tile ***********************************/
+        for n in range_stepped(intra_warp_unit_idx, CMMA_FILTER_TILE_SIZE, WARP_SIZE) {
+            // Compute where in the slice we are starting
+            let rel_slice_row = n / CMMA_K; // Relative row (0 - 15)
+            let abs_slice_row = k + rel_slice_row; // Row of the matrix the slice is on
+            let abs_slice_col = b_col + (n % 16); // Row of the matrix the slice is on
+
+            // Given the row of the matrix that the slice is in, and the index of the unit
+            // within a slice, want to compute what weight element to load...
+            let out_channel = abs_slice_col;
+            let in_channel = abs_slice_row / kernel_size;
+            let kernel_y = (abs_slice_row % kernel_size) / kernel_h;
+            let kernel_x = abs_slice_row % kernel_w;
+
+            weight_tile[n] = f16::cast_from(
+                weight[out_channel * weight.stride(0)
+                    + in_channel * weight.stride(1)
+                    + kernel_y * weight.stride(2)
+                    + kernel_x * weight.stride(3)],
+            );
+        }
+
+        /**************************** Bounds Check + CMMA Op*********************************/
+        if a_row < gemm_m && k < gemm_k && b_col < gemm_n {
+            cmma::load(&matrix_a, input_tile.as_slice(), CMMA_K);
```

Comment:
When doing the matmul, i found you can safely reuse a fragment for several executions. So this line could go in the outer loop.

Reply:
I'm not sure what you mean? I am reusing the fragments, but I still need to load a tile and execute the matmul for each `k`.